packages/in_app_purchase/in_app_purchase_android/lib/src/billing_client_wrappers/sku_details_wrapper.dart

```diff
@@ -32,7 +32,9 @@ class SkuDetailsWrapper {
     required this.description,
     required this.freeTrialPeriod,
     required this.introductoryPrice,
-    required this.introductoryPriceMicros,
+    @Deprecated('Use `introductoryPriceAmountMicros` parameter instead')
+        String introductoryPriceMicros = '',
+    this.introductoryPriceAmountMicros = 0,
```

Comment:
I think the better option here is to make this `int introductoryPriceAmountMicros,`, and have the constructor body do:
```
introductoryPriceAmountMicros = introductoryPriceAmountMicros ?? int.tryParse(introductoryPriceMicros) ?? 0
```

Then you don't need all the complicated logic in the getter, or `_introductoryPriceMicros`.

Reply:
We'll be changing it to required at that point presumably to match everything else, and it will already be a breaking change in general, so that doesn't seem like a significant issue. But it's not a big deal either way.